dvc/cli/__init__.py

```diff
@@ -48,6 +49,16 @@ def main(argv=None):  # noqa: C901
     from dvc.exceptions import DvcException, NotDvcRepoError
     from dvc.logger import FOOTER, disable_other_loggers
 
+    # NOTE: stderr/stdout may be closed if we are running from dvc.daemon.
+    # On Linux we directly call cli.main after double forking and closing
+    # the copied parent's standard file descriptors. If we make any logging
+    # calls in this state it will cause an exception due to writing to a closed
+    # file descriptor.
+    if sys.stderr.closed:  # pylint: disable=using-constant-test
+        logging.disable()
+    elif sys.stdout.closed:  # pylint: disable=using-constant-test
+        logging.disable(logging.INFO)
```

Comment:
What do you think of reducing the scope of this to only when we are running in the daemon mode, i.e. check if `DVC_DAEMON` is set?

Reply:
Technically, users could run DVC in background processes in this state (without using `dvc daemon`), so I guess it depends on whether or not we think that should be an error condition.